test/e2e/e2e_test.go

```diff
@@ -118,83 +104,221 @@ func TestE2EFlow(t *testing.T) {
-		err = awaitility.Client.Delete(context.TODO(), currentMur)
+	// 0. Verify that the `basic` NSTemplateTier resource exists (will be needed later)
+	basicTier := v1alpha1.NSTemplateTier{}
+	err = awaitility.Host().Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: awaitility.HostNs,
+		Name:      "basic",
+	}, &basicTier)
+	if err != nil {
+		return nil, nil, err
+	}
+	revisions := make(map[string]string, 3)
+	for _, typ := range []string{"code", "dev", "stage"} {
+		if r, found := namespaceRevision(basicTier, typ); found {
+			revisions[typ] = r
+			continue
+		}
+		return nil, nil, fmt.Errorf("unable to find revision for '%s' namespace in the 'basic' NSTemplateTier", typ)
+	}
 
-		// then
-		require.NoError(t, err)
-		t.Logf("MasterUserRecord '%s' deleted", mur.Name)
+	// 1. Create a UserSignup resource
+	userID := uuid.NewV4().String()
+	userSignup, err := newUserSignup(awaitility.Host(), userID, username, username)
+	if err != nil {
+		return nil, nil, err
+	}
+	err = awaitility.Host().Client.Create(context.TODO(), userSignup, testsupport.CleanupOptions(ctx))
+	if err != nil {
+		return nil, nil, err
+	}
+	// at this stage, the usersignup should not be approved nor completed
+	userSignup, err = awaitility.Host().WaitForUserSignup(userSignup.Name,
+		wait.UntilUserSignupHasConditions(
+			v1alpha1.Condition{
+				Type:   v1alpha1.UserSignupApproved,
+				Status: corev1.ConditionFalse,
+				Reason: "PendingApproval",
+			},
+			v1alpha1.Condition{
+				Type:   v1alpha1.UserSignupComplete,
+				Status: corev1.ConditionFalse,
+				Reason: "PendingApproval",
+			}))
+	if err != nil {
+		return nil, nil, err
+	}
 
-		verifyDeletion(t, awaitility, currentMur)
+	// 2. approve the UserSignup
+	userSignup.Spec.Approved = true
+	err = awaitility.Host().Client.Update(context.TODO(), userSignup)
+	if err != nil {
+		return nil, nil, err
+	}
+	// Check the updated conditions
+	_, err = awaitility.Host().WaitForUserSignup(userSignup.Name,
+		wait.UntilUserSignupHasConditions(
+			v1alpha1.Condition{
+				Type:   v1alpha1.UserSignupApproved,
+				Status: corev1.ConditionTrue,
+				Reason: "ApprovedByAdmin",
+			},
+			v1alpha1.Condition{
+				Type:   v1alpha1.UserSignupComplete,
+				Status: corev1.ConditionTrue,
+			}))
+	if err != nil {
+		return nil, nil, err
+	}
 
-		extraMur, err = wait.NewHostAwaitility(awaitility).WaitForMasterUserRecord(extraMur.Name)
-		require.NoError(t, err)
-		verifyResources(t, awaitility, extraMur.Name,
-			wait.UntilUserAccountHasConditions(isProvisioned()))
-	})
+	// 3. Wait for MUR to be created
+	mur, err := awaitility.Host().WaitForMasterUserRecord(username,
+		wait.UntilMasterUserRecordHasConditions(isProvisioned()),
+		wait.UntilMasterUserRecordHasUserAccountStatuses(
+			v1alpha1.UserAccountStatusEmbedded{
+				TargetCluster: memberCluster.Name,
+				UserAccountStatus: toolchainv1alpha1.UserAccountStatus{
+					Conditions: []toolchainv1alpha1.Condition{
+						isProvisioned(),
+					},
+				},
+			},
```

Comment:
you don't set the `SyncIndex`. I guess that is also the reason why it is failing: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/codeready-toolchain_toolchain-e2e/30/pull-ci-codeready-toolchain-toolchain-e2e-master-e2e/278/build-log.txt

Reply:
I need to dig that further, because e2e tests pass on a temp cluster...

Reply:
oh anyways, with the refactoring you suggested, this is going to be removed...